open-api/rest-catalog-open-api.yaml

```diff
@@ -1903,6 +1926,34 @@ components:
       schema:
         type: string
 
+    idempotency-key:
+      name: Idempotency-Key
+      in: header
+      required: false
+      schema:
+        type: string
+        format: uuid
+        minLength: 36
+        maxLength: 36
+        example: "550e8400-e29b-41d4-a716-446655440000"
+      description: |
+        Optional client-provided idempotency key for safe request retries.
+
+        When present, the server ensures no additional effects for requests that carry the same
+        Idempotency-Key within the same operation/resource scope. If a prior request with this key
+        has been finalized, the server returns the previously finalized response instead of
+        re-executing the mutation.
```

Comment:
For some endpoints, such as `updateTable`, the response payload can be large (a full Metadata JSON). In order to implement the behavior described here, the Catalog would have to store the potentially large payload for each commit in conjunction with the otherwise lightweight idempotency key.

Wouldn't the information "already processed" be enough for a client? What is the motivation to return the full previous response?

Reply:
Reworded. Thanks